include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,550 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser.
+ */
+
+#pragma once
+
+#include <deque>
+
+#include <seqan3/range/hash.hpp>
+
+namespace seqan3::detail
+{
+// ---------------------------------------------------------------------------------------------------------------------
+// minimiser class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by seqan3::views::minimiser.
+ * \tparam urng_t The type of the underlying ranges, must model std::forward_range, the reference type must model
+ *                std::size_t. The typical use case is that the reference type is the result of seqan3::kmer_hash.
+ * \implements std::ranges::view
+ * \implements std::ranges::random_access_range
+ * \implements std::ranges::sized_range
+ * \ingroup views
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::view urng_t>
+class minimiser : public std::ranges::view_interface<minimiser<urng_t>>
+{
+private:
+    static_assert(std::ranges::forward_range<urng_t const>, "The minimiser only works on forward_ranges");
+
+    //!\brief The underlying range.
+    urng_t urange;
+
+    //!\brief The number of elements in one window.
+    uint32_t w_elems;
```

Comment:
I think the terms window span/size/length would be misleading for someone who reads about minimizer in papers, because there these terms refer to the window on the text not on the hash values, so I think another term makes it less confusing, but I will change it to num_w_elems. Thanks! :)